components/suggest/src/rs.rs

```diff
@@ -81,7 +91,7 @@ impl SuggestRemoteSettingsClient for remote_settings::Client {
 /// suggestions are stored in each record's attachment.
 #[derive(Clone, Debug, Deserialize, PartialEq)]
 #[serde(tag = "type")]
-pub(crate) enum SuggestRecord {
+pub enum SuggestRecord {
```

Comment:
One other reason I wanted to suggest (😝) using `SuggestionProvider` instead of `SuggestRecord` is that we use `SuggestRecord` to parse Remote Settings record payloads.

Right now, this is OK, because the records are empty (save for the `type` tag), and none of the variants have associated data. But this won't work once we add associated data to variants, like #6089—because the weather prefixes are stored in the record, not in the attachment.